sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_client_async.py

```diff
@@ -980,12 +991,44 @@ async def settle_messages_async(
         except TypeError:
             first = delivery_id
             last = None
+
+        # Create a Message Delivery object for the Disposition
+        message_delivery = _MessageDelivery(None, MessageDeliveryState.WaitingToBeSent, expire_time)
+        on_disposition_received = partial(self._on_disposition_received_async, message_delivery)
+
         await self._link.send_disposition(
             first_delivery_id=first,
             last_delivery_id=last,
             delivery_tag=delivery_tag,
-            settled=True,
+            settled=False,
             delivery_state=state,
             batchable=batchable,
             wait=True,
+            on_disposition=on_disposition_received,
         )
+
+        # Wait for the message to be settled
+        running = True
+        while running and message_delivery.state not in MESSAGE_DELIVERY_DONE_STATES:
+            # TODO: Confirm if this is the correct way to handle the do_work call
+            # calling do_work directly triggers Flow() calls which is not what we want
+            running = await self.do_work_async(flow=False)
```

Comment:
is this where we typically use `connection.listen` ?

Reply:
yup but I want to not send a flow which could happen after a drain b/c of _client_run_async -- some investigating can happen here